src/vues/stock.js

```diff
@@ -1,48 +1,57 @@
-import { uneVueContenant } from './contenant.js';
+import { VueContenant } from './contenant.js';
+import { VueContenu } from './contenu.js';
 const imageEtageres = require('../images/stock.png');
 
-class VueStock {
-  constructor (topologie) {
+export class VueStock {
+  constructor (pointInsertion, topologie) {
     this.topologie = topologie;
+    this.stock = document.createElement('div');
+    this.stock.id = 'stock';
+    this.stock.classList.add('stock');
+    document.getElementById(pointInsertion).appendChild(this.stock);
   }
 
-  affiche (idMagasin, contenants) {
-    const vue = this;
+  creerElementContenants (dimensionsEtageres) {
+    let elementContenants = document.createElement('div');
+    elementContenants.classList.add('contenants');
+    elementContenants.id = 'contenants';
+    this.stock.appendChild(elementContenants);
+
+    const resizeContenants = function () {
+      elementContenants.style.width = dimensionsEtageres.width + 'px';
+      elementContenants.style.height = dimensionsEtageres.height + 'px';
+    };
+
+    resizeContenants();
+    window.addEventListener('resize', resizeContenants);
 
-    const magasin = document.getElementById(idMagasin);
+    return elementContenants;
+  }
+
+  affiche (contenants) {
+    const vue = this;
```

Comment:
Je pense que ES2015 nous permet de capturer la valeur de `this` dans une fermeture avec la construction `() => { … }`. Je propose d'utiliser cette construction plutôt que de passer par une constante `vue`.

Reply:
👍 👍